packages/fuels-signers/src/provider.rs

```diff
@@ -15,29 +16,25 @@ use fuel_gql_client::{
 use std::collections::HashMap;
 use thiserror::Error;
 
-use crate::wallet::WalletError;
 use fuels_core::parameters::TxParameters;
 use fuels_types::errors::Error;
 
-/// An error involving a signature.
 #[derive(Debug, Error)]
 pub enum ProviderError {
-    #[error("Request failed: {0}")]
-    TransactionRequestError(String),
+    // Every IO error in the context of Provider comes from the gql client
     #[error(transparent)]
     ClientRequestError(#[from] io::Error),
-    #[error("Wallet error: {0}")]
-    WalletError(String),
 }
 
-impl From<WalletError> for ProviderError {
-    fn from(e: WalletError) -> Self {
-        ProviderError::WalletError(e.to_string())
+impl From<ProviderError> for Error {
+    fn from(e: ProviderError) -> Self {
+        Error::ProviderError(e.to_string())
     }
 }
+
 /// Encapsulates common client operations in the SDK.
 /// Note that you may also use `client`, which is an instance
-/// of `FuelClient`, directly, which providers a broader API.
+/// of `FuelClient`, directly, which provides a broader API.
```

Comment:
Should we keep this comment? I feel like it's a little misguiding, we are explicitely doing these changes so we stop using the `FuelClient` directly. wdyt?

Reply:
Since we don't fully wrap the client I figured we should still leave the user a way to interact with it. 
@segfault-magnet also made a good point about the possibility that the client could add new functionalities before the SDK catches up so the user would still have some way to access those until our next release.